src/main/java/model/member/ExtendedBattleMember.java

```diff
@@ -484,56 +486,46 @@ protected boolean checkRequirements(final Talent talent) {
     }
 
     protected boolean checkRequirements(final Item item) {
-        String[] parsed = item.getRequirement().split(" ");
+        List<Character> requirementsList = new ArrayList<>();
+        for (char c : item.getRequirement().toCharArray()) {
+            requirementsList.add(c);
+        }
 
-        for (int i = 0; i < parsed.length; i += 2) {
-            if (NumberUtils.isDigits(parsed[i])) {
-                if (i + 1 >= parsed.length) {
+        while (!requirementsList.isEmpty()) {
+            int x = Utility.parseNumber(requirementsList);
+            String attribute = Utility.parseString(requirementsList);
+
+            if (attribute.equalsIgnoreCase(PrimaryAttribute.strength.toShortString())) {
+                if (x > getStrength()) {
```

Comment:
What is going on here?
getRequirement() returns a String, which you convert to a charArray. You then put all array elements in a list.
After that, you parse,, while the list is not empty, numbers and attributes from the list?

So we have a string like "5 KK 3 BW" I imagine?
This is beyond ugly. Especially, that the parseNumber or parseString seems to remove elements from the list

Reply:
Yes, I renamed the methods to consumeNumber.
I have no better solution at the moment